File: ExchangeInterface/cssParser.cs

namespace LaTrompa.WebCrap
{

    
    // Pfff, this "parser" is pure CRAP, 
    // but all the rest is either too big, also shit, and/or f*ing using LINQ...
    // From https://gist.github.com/hgarcia/561823
    public class CssParser
    {
        private System.Collections.Generic.List<string> _styleSheets;
        private System.Collections.Generic.SortedList<string, StyleClass> _scc;
        
        public System.Collections.Generic.SortedList<string, StyleClass> Styles
        {
            get { return this._scc; }
            set { this._scc = value; }
        } // End Property Styles


        public StyleClass InlineStyles
        {
            get { return this._scc["inline"]; }
        } // End Read-Only Property InlineStyles


        public CssParser() : this("")
        { }


        public CssParser(string content)
        {
            this._styleSheets = new System.Collections.Generic.List<string>();
            this._scc = new System.Collections.Generic.SortedList<string, StyleClass>();

            if (!string.IsNullOrEmpty(content))
            {
                this.AddInlineStyle(content);
            } // End if (!string.IsNullOrEmpty(content))
        } // End Constructor


        public void AddStyleSheet(string path)
        {
            this._styleSheets.Add(path);
            ProcessStyleSheet(path);
        }


        public void AddInlineStyle(string content)
        {
            this._styleSheets.Add("inline");

            ProcessStyleSheetFromString("inline { " + content + " } ");
        } // End Sub AddInlineStyle 


        public string GetStyleSheet(int index)
        {
            return this._styleSheets[index];
        }


        private void ProcessStyleSheetFromString(string content)
        {
            string strContent = CleanUp(content);
            string[] parts = strContent.Split('}');
            foreach (string s in parts)
            {
                if (CleanUp(s).IndexOf('{') > -1)
                {
                    FillStyleClass(s);
                }
            }
        }


        private void ProcessStyleSheet(string path)
        {
            string content = CleanUp(System.IO.File.ReadAllText(path));
            string[] parts = content.Split('}');
            foreach (string s in parts)
            {
                if (CleanUp(s).IndexOf('{') > -1)
                {
                    FillStyleClass(s);
                }
            }
        }


        private void FillStyleClass(string s)
        {
            StyleClass sc = null;
            string[] parts = s.Split('{');
            string styleName = CleanUp(parts[0]).Trim().ToLower();

            if (this._scc.ContainsKey(styleName))
            {
                sc = this._scc[styleName];
                this._scc.Remove(styleName);
            }
            else
            {
                sc = new StyleClass();
            }

            sc.Name = styleName;

            string[] atrs = CleanUp(parts[1]).Replace("}", "").Split(';');
            foreach (string a in atrs)
            {
                if (a.Contains(":"))
                {
                    string _key = a.Split(':')[0].Trim().ToLower();
                    if (sc.Attributes.ContainsKey(_key))
                    {
                        sc.Attributes.Remove(_key);
                    }
                    sc.Attributes.Add(_key, a.Split(':')[1].Trim().ToLower());
                }
            }
            this._scc.Add(sc.Name, sc);
        }


        private string CleanUp(string s)
        {
            string temp = s;
            string reg = "(/\\*(.|[\r\n])*?\\*/)|(//.*)";
            System.Text.RegularExpressions.Regex r = new System.Text.RegularExpressions.Regex(reg);
            temp = r.Replace(temp, "");
            temp = temp.Replace("\r", "").Replace("\n", "");
            return temp;
        }


    } // End Class CssParser


    public class StyleClass
    {
        private string _name = string.Empty;
        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        private System.Collections.Generic.SortedList<string, string> _attributes = new System.Collections.Generic.SortedList<string, string>();
        public System.Collections.Generic.SortedList<string, string> Attributes
        {
            get { return _attributes; }
            set { _attributes = value; }
        }


        public override string ToString()
        {
            string strRetValue = null;
            System.Text.StringBuilder sb = new System.Text.StringBuilder();

            foreach (System.Collections.Generic.KeyValuePair<string, string> kvp in Attributes)
            {
                sb.Append(kvp.Key + ":" + kvp.Value + ";");
            } // Next kvp

            strRetValue = sb.ToString();
            sb.Length = 0;
            sb = null;

            return strRetValue;
        } // End Function ToString


    } // End Class StyleClass


} // End Namespace LaTrompa.WebCrap

File: SvgXmlBeautifierVb/cssParser.vb

Namespace LaTrompa.WebCrap


    ' Pfff, this "parser" is pure CRAP, 
    ' but all the rest is either too big, also crap, or using LINQ, 
    ' which doesn't work in .NET 2.0
    ' From https://gist.github.com/hgarcia/561823
    Public Class CssParser
        Private _styleSheets As System.Collections.Generic.List(Of String)
        Private _scc As System.Collections.Generic.SortedList(Of String, StyleClass)


        Public Property Styles() As System.Collections.Generic.SortedList(Of String, StyleClass)
            Get
                Return Me._scc
            End Get
            Set(value As System.Collections.Generic.SortedList(Of String, StyleClass))
                Me._scc = value
            End Set
        End Property ' Styles


        Public ReadOnly Property InlineStyles() As StyleClass
            Get
                Return Me._scc("inline")
            End Get
        End Property ' Read-Only InlineStyles 


        Public Sub New()
            Me.New("")
        End Sub


        Public Sub New(content As String)
            Me._styleSheets = New System.Collections.Generic.List(Of String)()
            Me._scc = New System.Collections.Generic.SortedList(Of String, StyleClass)()

            If Not String.IsNullOrEmpty(content) Then
                Me.AddInlineStyle(content)
                ' End if (!string.IsNullOrEmpty(content))
            End If
        End Sub ' End Constructor


        Public Sub AddStyleSheet(path As String)
            Me._styleSheets.Add(path)
            ProcessStyleSheet(path)
        End Sub


        Public Sub AddInlineStyle(content As String)
            Me._styleSheets.Add("inline")

            ProcessStyleSheetFromString((Convert.ToString("inline { ") & content) + " } ")
        End Sub ' AddInlineStyle 


        Public Function GetStyleSheet(index As Integer) As String
            Return Me._styleSheets(index)
        End Function


        Private Sub ProcessStyleSheetFromString(content As String)
            Dim strContent As String = CleanUp(content)
            Dim parts As String() = strContent.Split("}"c)
            For Each s As String In parts
                If CleanUp(s).IndexOf("{"c) > -1 Then
                    FillStyleClass(s)
                End If
            Next
        End Sub


        Private Sub ProcessStyleSheet(path As String)
            Dim content As String = CleanUp(System.IO.File.ReadAllText(path))
            Dim parts As String() = content.Split("}"c)
            For Each s As String In parts
                If CleanUp(s).IndexOf("{"c) > -1 Then
                    FillStyleClass(s)
                End If
            Next
        End Sub


        Private Sub FillStyleClass(s As String)
            Dim sc As StyleClass = Nothing
            Dim parts As String() = s.Split("{"c)
            Dim styleName As String = CleanUp(parts(0)).Trim().ToLower()

            If Me._scc.ContainsKey(styleName) Then
                sc = Me._scc(styleName)
                Me._scc.Remove(styleName)
            Else
                sc = New StyleClass()
            End If

            sc.Name = styleName

            Dim atrs As String() = CleanUp(parts(1)).Replace("}", "").Split(";"c)
            For Each a As String In atrs
                If a.Contains(":") Then
                    Dim _key As String = a.Split(":"c)(0).Trim().ToLower()
                    If sc.Attributes.ContainsKey(_key) Then
                        sc.Attributes.Remove(_key)
                    End If
                    sc.Attributes.Add(_key, a.Split(":"c)(1).Trim().ToLower())
                End If
            Next
            Me._scc.Add(sc.Name, sc)
        End Sub


        Private Function CleanUp(s As String) As String
            Dim temp As String = s
            Dim reg As String = "(/\*(.|[" & vbCr & vbLf & "])*?\*/)|(//.*)"
            Dim r As New System.Text.RegularExpressions.Regex(reg)
            temp = r.Replace(temp, "")
            temp = temp.Replace(vbCr, "").Replace(vbLf, "")
            Return temp
        End Function


    End Class ' CssParser


    Public Class StyleClass
        Private _name As String = String.Empty
        Public Property Name() As String
            Get
                Return _name
            End Get
            Set(value As String)
                _name = value
            End Set
        End Property

        Private _attributes As New System.Collections.Generic.SortedList(Of String, String)()
        Public Property Attributes() As System.Collections.Generic.SortedList(Of String, String)
            Get
                Return _attributes
            End Get
            Set(value As System.Collections.Generic.SortedList(Of String, String))
                _attributes = value
            End Set
        End Property


        Public Overrides Function ToString() As String
            Dim strRetValue As String = Nothing
            Dim sb As New System.Text.StringBuilder()

            For Each kvp As System.Collections.Generic.KeyValuePair(Of String, String) In Attributes
                sb.Append(kvp.Key + ":" + kvp.Value + ";")
            Next
            ' Next kvp
            strRetValue = sb.ToString()
            sb.Length = 0
            sb = Nothing

            Return strRetValue
        End Function ' ToString


    End Class ' StyleClass


End Namespace ' LaTrompa.WebCrap
